pkg/build/builder/daemonless.go

```diff
@@ -313,10 +301,77 @@ func buildDaemonlessImage(sc types.SystemContext, store storage.Store, isolation
 		DropCapabilities:        dropCapabilities(),
 	}
 
-	_, _, err := imagebuildah.BuildDockerfiles(opts.Context, store, options, opts.Dockerfile)
+	_, _, err = imagebuildah.BuildDockerfiles(opts.Context, store, options, opts.Dockerfile)
 	return err
 }
 
+func generateTransientMounts() ([]string, error) {
+	mounts := []string{}
+	mounts, err := appendRHSMMount(mounts)
+	if err != nil {
+		return mounts, err
+	}
+	mounts, err = appendCATrustMount(mounts)
+	if err != nil {
+		return mounts, err
+	}
+	return mounts, nil
+}
+
+func appendRHSMMount(mounts []string) ([]string, error) {
+	st, err := os.Stat("/run/secrets/rhsm")
+	if err != nil {
+		return mounts, errors.Wrap(err, "failed to stat /run/secrets/rhsm")
+	}
+	if !st.IsDir() {
+		return mounts, nil
+	}
+
+	// Add a bind of /run/secrets/rhsm, to pass along anything that the
+	// runtime mounted from the node into our /run/secrets/rhsm.
+	log.V(0).Infof("Adding transient rw bind mount for /run/secrets/rhsm")
+	tmpDir, err := ioutil.TempDir("/tmp", "rhsm-copy")
+	if err != nil {
+		return mounts, errors.Wrap(err, "failed to create tmpdir for /run/secrets/rhsm")
+	}
+	fs := s2ifs.NewFileSystem()
+	err = fs.CopyContents("/run/secrets/rhsm", tmpDir, map[string]string{})
+	if err != nil {
+		return mounts, err
+	}
+	mounts = append(mounts, fmt.Sprintf("%s:/run/secrets/rhsm:rw,nodev,noexec,nosuid", tmpDir))
+	return mounts, nil
+}
+
+func appendCATrustMount(mounts []string) ([]string, error) {
+	mountCAEnv, exists := os.LookupEnv("BUILD_MOUNT_ETC_PKI_CATRUST")
+	if !exists {
+		return mounts, nil
+	}
+
+	mountCA, err := strconv.ParseBool(mountCAEnv)
+	if err != nil {
+		log.V(5).Infof("failed to parse BUILD_MOUNT_ETC_PKI_CATRUST: %v", err)
+		return mounts, errors.Wrap(err, "failed to parse BUILD_MOUNT_ETC_PKI_CATRUST")
```

Comment:
Likewise I don't think we should fail a build if we can't parse this env var or we don't find the trust bundle. If a build depends on this information, it will eventually fail.